examples/jsm/csm/Shader.js

```diff
@@ -78,24 +78,64 @@ IncidentLight directLight;
 	DirectionalLightShadow directionalLightShadow;
 	#endif
 
+	#if defined( CSM_FADE )
+	for ( int i = 0; i < NUM_DIR_LIGHTS; i ++ ) {
+
+		directionalLight = directionalLights[ i ];
+		getDirectionalDirectLightIrradiance( directionalLight, geometry, directLight );
+
+		float margin = 0.25 * pow( linearDepth, 2.0 );
+		float csmx = CSM_cascades[ i ].x - margin / 2.0;
+		float csmy = CSM_cascades[ i ].y + margin / 2.0;
+		if( i < NUM_DIR_LIGHT_SHADOWS && linearDepth >= csmx && ( linearDepth < csmy ||  i == CSM_CASCADES - 1 ) ) {
```

Comment:
This is the case that prompted #18751.

It would be nice to remove the below `if(i < NUM_DIR_LIGHT_SHADOWS) {` block by using `#if ( UNROLLED_LOOP_INDEX < NUM_DIR_LIGHT_SHADOWS )` . Are there any suggestions for how to apply `#pragma unroll_loop` here?